authzed/api/v1/error_reason.proto

```diff
@@ -268,4 +268,21 @@ enum ErrorReason {
   //       }
   //     }
   ERROR_REASON_ATTEMPT_TO_RECREATE_RELATIONSHIP = 18;
+
+  // The request failed because it caused the maximum depth allowed to be
+  // exceeded. This typically indicates that there is a circular data traversal
+  // somewhere in the schema, but can also be raised if the data traversal is simply
+  // too deep.
```

Comment:
Maybe mention there's a flag to increase this limit?

Reply:
The error message itself will point to a docs page mentioning the flag